### PR TITLE
Adds Biofoam and Biofoam Injector

### LINKED ISSUE #10201: Biofoam
[Discussion]

I intend to code this at some point, this is mainly a note to self.

Biofoam injectors that, when used, can stem internal bleeding for five minutes by filling the body cavity with biodegradable foam.
Should make internal bleeding less of a guaranteed death sentence by giving the victim time to make it to surgery to have it repaired.

### PR DESCRIPTION
**Pull request by Shadowmech88**
biofoam_injector_full_display biofoam_injector_empty_display

I figure a couple of these could be nice for paramedics.
I'll add them to the map in another PR.

Resolves #10201

:cl:
 * rscadd: Adds biofoam, a chem that stops internal wounds from getting worse and completely stops internal bleeding while in your system.
 * rscadd: Adds biofoam injectors, a single-use item which injects enough biofoam into a patient to stop internal bleeding for five minutes.

### CODE REVIEW

**Comment by SonixApache** [comment]
If I loot the paramedic office and inject myself with all the biofoam, can I run around in space without popping lungs?

**Comment by Shadowmech88** [comment]
You'll still pop your lungs, but you won't bleed out from it until the biofoam runs out.

**Comment by jakmak3** [comment]
can biofoam be manually made or is it in roundstart/cargo order medikits only?

**Comment by Shadowmech88** [comment]
Can't be made.

_Currently_ it's unobtainable, in a second PR I'll actually add it to the Paramedic offices, in which case it'll be roundstart only.

**Comment by kol1th** [comment]
@Shadowmech88 Orderable by cargo, perhaps?

**Comment by Shadowmech88** [comment]
My only concern with that is that Medbay will order a ton of them and completely trivialize internal bleeding.
The item is really just meant for Paramedics to use a couple of times a round to keep patients from bleeding out until they can be brought to Medbay and get emergency surgery.

**Comment by jakmak3** [comment]
@shadowmech88 you overestimate the competence of the common medbay player. Even if they were willing to the work that hard it would be 20 times more effective to make a nanobot odysseys

**Comment by Sprok0** [comment]
This is also a 15u autoinjector, for chloraling even people who have antitox in their system.

**Comment by SonixApache** [comment]
Should probably make it so you can't extract the foam from these, unless you want 15u autoinjectors and/or infinite reagents through one of the many duplication methods

**Comment by Shadowmech88** [comment]
@Sprok0 
@SonixApache 

Reagents can neither be removed from nor added to this item through any means.

**Comment by coldcola** [comment]
sounds good as a paramedic item, what other methods than the oddy can dupe reagents after the dropper fix anyway?

**Comment by Shadowmech88** [comment]
Service borgs can duplicate reagents using their shaker.

**Comment by coldcola** [comment]
working as intended though, no?

**Comment by Shadowmech88** [comment]
Yes, it's intended.

**Comment by clusterfack** [comment]
I don't care at all myself, go emoji vote

**Comment by Intigracy** [comment]
You could add biofoam to one of the trader plants.

Not in this PR obviously.

**Comment by Shadowmech88** [comment]
The issue with making it obtainable outside the injectors is that it opens it up to oddy/service borg duplication.

As useful as this reagent is, I'm worried about the station becoming saturated with it.

I'll make a discussion issue for it after people get a chance to play around with the reagent in injector form.

**Comment by Exxion** [comment]
It's still open to borg duplication even when it's only obtainable in the injector. It's just more difficult.

**Comment by Shadowmech88** [comment]
I didn't think there was any way to extract reagents from the blood once injected.

**Comment by Exxion** [comment]
1. Vomit
2. IV pole (in very, very small amounts)

**Comment by Intigracy** [comment]
"Reagents can neither be removed from nor added to this item through any means."

I'm gonna have to ask you to take a syringe and try drawing from the autoinjector, because base autoinjectors let you do so, though it's weird and only lets you remove 5 units before it counts as empty and only put in 5 units before it counts as full.

**Comment by Shadowmech88** [comment]
I already have.

You can only use syringes on hyposprays and autoinjectors because they have the OPENCONTAINER flag, which I removed from this item.

Also autoinjectors have a capacity of 5u because their volume is set to 5u.

File: code/modules/reagents/reagent_containers/hypospray.dm [hunk 1]
```diff
@@ -98,6 +98,26 @@
...
+	name = "biofoam injector"
+	desc = "A small, single-use device used to administer biofoam in the field."
+	icon_state = "biofoam1"
+	amount_per_transfer_from_this = 15
+	volume = 15
+	flags = FPRINT
+
+/obj/item/weapon/reagent_containers/hypospray/autoinjector/biofoam_injector/New()
+	..()
+	reagents.remove_reagent(DOCTORSDELIGHT, 30)
```
Comment (Intigracy) [inline code/modules/reagents/reagent_containers/hypospray.dm:115]: then why is this line 30?

Comment (Shadowmech88) [reply code/modules/reagents/reagent_containers/hypospray.dm:115]: Copied it from `/obj/item/weapon/reagent_containers/hypospray/creatine/New()` and didn't bother to change it. It being 30 doesn't make a difference.

**Comment by Durgen** [comment]
how about giving biofoam a really complex recipe similar to mednanos

**Pull request merged**